Maven RT fails if invoked without project (mojo not requiring project)

Reproducer: Invoke this below in directory without POM
```
$ mvn eu.maveniverse.maven.plugins:toolbox:gav-search -Dexpression=jetty -e
```

As in those cases there is only "stub" MavenProject, but MIMA blindly wants values from project.

Throws this:
```
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal eu.maveniverse.maven.plugins:toolbox:0.1.1:gav-search failed: Cannot invoke "java.io.File.toPath()" because the return value of "org.apache.maven.project.MavenProject.getBasedir()" is null
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.toPath()" because the return value of "org.apache.maven.project.MavenProject.getBasedir()" is null
    at eu.maveniverse.maven.mima.runtime.maven.MavenRuntime.create (MavenRuntime.java:64)
    at eu.maveniverse.maven.toolbox.plugin.MojoSupport.lambda$execute$3 (MojoSupport.java:455)
    at eu.maveniverse.maven.toolbox.plugin.MojoSupport.lambda$getOrCreate$0 (MojoSupport.java:137)
    at java.util.concurrent.ConcurrentHashMap.computeIfAbsent (ConcurrentHashMap.java:1740)
    at eu.maveniverse.maven.toolbox.plugin.MojoSupport.getOrCreate (MojoSupport.java:137)
    at eu.maveniverse.maven.toolbox.plugin.MojoSupport.execute (MojoSupport.java:455)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
```

--- a/runtime/embedded-maven/src/main/java/eu/maveniverse/maven/mima/runtime/maven/MavenRuntime.java
+++ b/runtime/embedded-maven/src/main/java/eu/maveniverse/maven/mima/runtime/maven/MavenRuntime.java
@@ -24,7 +24,6 @@ import javax.inject.Provider;
 import javax.inject.Singleton;
 import org.apache.maven.execution.MavenExecutionRequest;
 import org.apache.maven.execution.MavenSession;
-import org.apache.maven.project.MavenProject;
 import org.apache.maven.rtinfo.RuntimeInformation;
 import org.apache.maven.settings.Proxy;
 import org.eclipse.aether.RepositorySystem;
@@ -65,18 +64,23 @@ public final class MavenRuntime extends RuntimeSupport {
     @Override
     public Context create(ContextOverrides overrides) {
         MavenSession mavenSession = mavenSessionProvider.get();
+        boolean projectPresent = mavenSession.getRequest().isProjectPresent();
 
-        MavenProject currentProject = mavenSession.getCurrentProject();
-        Path basedir =
-                currentProject != null ? currentProject.getBasedir().toPath().toAbsolutePath() : DEFAULT_BASEDIR;
+        Path basedir;
+        if (projectPresent) {
+            basedir = mavenSession.getCurrentProject().getBasedir().toPath().toAbsolutePath();
+        } else {
+            basedir = DEFAULT_BASEDIR;
+        }
         MavenUserHome mavenUserHome = discoverMavenUserHome(mavenSession.getRequest());
         MavenSystemHome mavenSystemHome = discoverMavenSystemHome(mavenSession.getRequest());
         RepositorySystemSession session = mavenSession.getRepositorySession();
 
         ContextOverrides.Builder effectiveOverridesBuilder = overrides.toBuilder();
         effectiveOverridesBuilder.withUserSettings(true); // embedded
-        if (currentProject != null) {
-            effectiveOverridesBuilder.repositories(currentProject.getRemoteProjectRepositories());
+        if (projectPresent) {
+            effectiveOverridesBuilder.repositories(
+                    mavenSession.getCurrentProject().getRemoteProjectRepositories());
         }
         effectiveOverridesBuilder.systemProperties(session.getSystemProperties());
         effectiveOverridesBuilder.userProperties(session.getUserProperties());

